### PR TITLE
cursor: added support for gamepad sticks

### PR DESCRIPTION
**Pull request by larsiusprime**

### CODE REVIEW

File: flixel/addons/ui/FlxMultiGamepadAnalogStick.hx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	}
+	
+	private override function checkJustPressed():Bool
+	{
+		var value = false;
+		var dz = gamepad.deadZone;
+		switch(sInput.id)
+		{
+			case LEFT_ANALOG_STICK: if (sInput.axis == X)
+									{
```
Comment (Gama11) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: Something gone wrong with the indent here?

File: flixel/addons/ui/FlxMultiGamepadAnalogStick.hx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+	public function new(Gamepad:FlxGamepad, Input:StickInput, ?Combos:Array<FlxGamepadInputID>, ?Forbiddens:Array<FlxGamepadInputID>) 
+	{
+		sInput = Input;
+		super(Gamepad, Input.id, Combos, Forbiddens);
+	}
+	
+	override public function destroy():Void 
+	{
+		super.destroy();
+		gamepad = null;
```
Comment (Gama11) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: What gamepad? From the superclass?

Comment (larsiusprime) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: copypasta error looks like.

File: flixel/addons/ui/FlxMultiGamepadAnalogStick.hx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	}
+	
+	private override function checkJustPressed():Bool
+	{
+		var value = false;
+		var dz = gamepad.deadZone;
+		switch(sInput.id)
```
Comment (Gama11) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: I think these could technically be `return-switch`es too? If you like it better this way though keep it.

Comment (larsiusprime) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: Can you even do a return-switch if you have if logic inside the case statements?

Comment (Gama11) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: You can combine expressions pretty much however you want, and in Haxe, everything is an expression (which is why return-switch, return-if-else, return return return return value etc works). http://try.haxe.org/#62480 You just have to make sure there's always a non-Void value in each branch that can be assigned / returned.

Comment (larsiusprime) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: I see, so you just end on 
if(whatever)
{
  true;
}

Comment (Gama11) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: You'd need an else for each if, otherwise there are "Void branches".

Comment (MSGhero) [inline flixel/addons/ui/FlxMultiGamepadAnalogStick.hx]: I've noticed times when putting the value on the last line of a case doesn't work. Maybe I have too much logic or something, but you can put an additional return if it doesn't work.
```
return switch(thing) {
     case 1:
          logic logic logic
          return true;
     case _: false;
}
```

**Comment by larsiusprime** [comment]
is something broken here?

**Comment by MSGhero** [comment]
All 4 of your pulls on flixel failed so it's probably something else.

**Pull request merged**